src/CsvWriter.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace Keboola\Csv;
+
+class CsvWriter extends AbstractCsvFile
+{
+    /**
+     * @var resource
+     */
+    protected $filePointer;
+
+    /**
+     * @var string
+     */
+    private $mode;
+
+    /**
+     * CsvFile constructor.
+     * @param string $fileName
+     * @param string $delimiter
+     * @param string $enclosure
+     * @param string $mode
+     * @throws InvalidArgumentException
+     */
+    public function __construct(
+        $fileName,
+        $delimiter = self::DEFAULT_DELIMITER,
+        $enclosure = self::DEFAULT_ENCLOSURE,
+        $mode = 'w'
+    ) {
+        parent::__construct($fileName);
+        $this->setMode($mode);
+        $this->setDelimiter($delimiter);
+        $this->setEnclosure($enclosure);
+        $this->openCsvFile();
+    }
+
+    public function __destruct()
+    {
+        $this->closeFile();
+    }
+
+    /**
+     * @param array $row
+     * @throws Exception
+     */
+    public function writeRow(array $row)
+    {
+        $str = $this->rowToStr($row);
+        $ret = fwrite($this->getFilePointer(), $str);
+
+        /* According to http://php.net/fwrite the fwrite() function
+         should return false on error. However not writing the full
+         string (which may occur e.g. when disk is full) is not considered
+         as an error. Therefore both conditions are necessary. */
+        if (($ret === false) || (($ret === 0) && (strlen($str) > 0))) {
+            throw new Exception(
+                "Cannot write to file {$this->getPathname()}",
+                Exception::WRITE_ERROR,
+                null,
+                Exception::WRITE_ERROR_STR
+            );
+        }
+    }
+
+    /**
+     * @param array $row
+     * @return string
+     * @throws Exception
+     */
+    public function rowToStr(array $row)
+    {
+        $return = [];
+        foreach ($row as $column) {
+            if (!(
+                is_scalar($column)
```

Comment:
Tady mě ještě teď když to vidím zformátované napadlo - nešlo by z toho udělat `if (!$stringable)` - to by neovlivnilo early exit v tom boolean výrazu a přitom by byla ta podmínka pak čitelnější.

Reply:
$isStringable
Diff 0 : 489.37952709198
Diff 1 : 497.38059091568
Diff 2 : 503.51376700401
Diff 3 : 505.61457920074
Diff 4 : 507.42060494423
Diff 5 : 504.46329498291
Diff 6 : 503.64453482628
Diff 7 : 501.33763980865
Diff 8 : 501.47956895828
Diff 9 : 497.10404992104

vs

soucasny stav
Diff 0 : 474.2037358284
Diff 1 : 463.99877309799
Diff 2 : 459.16944909096
Diff 3 : 457.29524803162
Diff 4 : 472.84506487846
Diff 5 : 491.63278388977
Diff 6 : 492.76865696907
Diff 7 : 489.76580095291
Diff 8 : 488.95872497559
Diff 9 : 478.8238589763

sice uz to neni takovej rozdil, ale je tam, takze ne, nebudu to prepisovat aby to bylo citelnejsi :) budes se muset naucit cist :trollface:

Reply:
njn :/